javascript/app-common/src/CommonLayout.tsx

```diff
@@ -18,16 +19,30 @@ import {
 import type { CommonLayoutProps } from './CommonLayout.types.js'
 
 export const CommonLayout: React.FC<CommonLayoutProps> = memo(
-	function CommonLayout({ children, configRail, detailRail, menu }) {
+	function CommonLayout({
+		children,
+		configRail,
+		detailRail,
+		menu,
+		panelsVisibility = true,
```

Comment:
In fact, because this is true by default, let's call it `hidePanels`, that way the prop can be defined without setting 

e.g. 
`<X hidePanels />`
vs `<X showPanels={false}`/>